storage/innobase/row/row0ins.cc

```diff
@@ -1328,6 +1328,8 @@ row_ins_foreign_check_on_constraint(
 		ib::info() << "WSREP: foreign key append failed: " <<  err;
 		goto nonstandard_exit_func;
 	}
+
+	wsrep_thd_set_PA_unsafe(trx->mysql_thd);
```

Comment:
Can we please modify the code in `wsrep_append_foreign_key()` instead of potentially slowing down all `INSERT` statements for everyone (in particular those who do not use Galera)?

There are two callers to `wsrep_append_foreign_key()`, so perhaps that interface may have to be simplified. The key thing is that there is no additional call if `!trx->is_wsrep()`.

In the other call, I see that we are evaluating some parameters in the caller. Could that logic be pushed down to the function itself, to avoid redundant comparisons of `wsrep_protocol_version` when Galera is not being used? We only need something so that the two `wsrep_append_foreign_key()` calls can be distinguished, and that this extra function call will be avoided for the other call of `wsrep_append_foreign_key()` (in `row_ins_check_foreign_constraint()`).